meson.build

```diff
@@ -23,142 +23,183 @@ project(
 
 lib_version = '0.5.0'
 
+# TODO: pkg-config
+# TODO: cmake-config
+
+inc_include = include_directories('include')
+
+#####################
+# Config Generation #
+#####################
+
 cc = meson.get_compiler('c')
+cc_native = meson.get_compiler('c', native: true)
 
-if ['gcc', 'clang'].contains(cc.get_id())
-  add_project_arguments(
-    # I've explicitly skipped the duplicated -W versions when they also test
-    # for the -Werror version
-    cc.get_supported_arguments(
-      '-Wshadow',
-      '-Wmissing-prototypes',
-      '-Wcast-align',
-      '-Werror=address',
-      '-Werror=strict-prototypes',
-      '-Werror=write-strings',
-      '-Werror=implicit-function-declaration',
-      '-Werror=pointer-arith',
-      '-Werror=declaration-after-statement',
-      '-Werror=return-type',
-      '-Werror=uninitialized',
-      '-Wimplicit-fallthrough',
-      '-Werror=strict-overflow',
-      '-Wstrict-overflow=2',
-      '-Wno-format-zero-length',
-      '-Wformat',
-      '-Werror=format-security',
-      '-Wno-gnu-zero-variadic-macro-arguments',
-      '-fno-common',
-    ),
-    language : ['c'],
-  )
-  # We can't test the build type, so we can' add -D_FORTIFY_SOURCE=2 here
-  if host_machine.system() == 'darwin'
-    if cc.has_argument('-Wno-deprecated-declarations')
-      add_project_arguments('-Wno-deprecated-declarations', language : ['c'])
+conf = configuration_data()
+conf_native = configuration_data()
+
+cc_dict = {
+    'compiler': cc,
+    'machine': host_machine,
+    'config': conf,
+    'native': false
+}
+
+cc_native_dict = {
+    'compiler': cc_native,
+    'machine': build_machine,
+    'config': conf_native,
+    'native': true
+}
+
+configurations = [cc_dict, cc_native_dict]
+
+foreach entry : configurations
+  compiler = entry.get('compiler')
+  config = entry.get('config')
+  n = entry.get('native')
```

Comment:
Nit: please name it `native` or `is_native` if `native` can't be used. Let's make it easier to read.

Reply:
Agreed here.